lib/core/unittest/CPackedBitVectorTest.cc

```diff
@@ -42,6 +44,200 @@ std::string toBitString(const T& v) {
     }
     return result.str();
 }
+
+class CPackedBitVectorInternals : public core::CPackedBitVector {
+public:
+    static std::size_t maximumOneByteRunLength() {
+        return MAXIMUM_ONE_BYTE_RUN_LENGTH;
+    }
+    static std::size_t maximumTwoByteRunLength() {
+        return MAXIMUM_TWO_BYTE_RUN_LENGTH;
+    }
+    static std::size_t maximumThreeByteRunLength() {
+        return MAXIMUM_THREE_BYTE_RUN_LENGTH;
+    }
+    static std::size_t maximumFourByteRunLength() {
+        return MAXIMUM_FOUR_BYTE_RUN_LENGTH;
+    }
+    static void appendRun(std::size_t runLength, std::uint8_t& lastRunBytes, TUInt8Vec& runLengthBytes) {
+        core::CPackedBitVector::appendRun(runLength, lastRunBytes, runLengthBytes);
+    }
+    static void extendLastRun(std::size_t runLength, std::uint8_t& runBytes, TUInt8Vec& runLengthBytes) {
+        core::CPackedBitVector::extendLastRun(runLength, runBytes, runLengthBytes);
+    }
+    static std::uint8_t bytes(std::size_t runLength) {
+        return core::CPackedBitVector::bytes(runLength);
+    }
+    static std::size_t readLastRunLength(std::uint8_t lastRunBytes,
+                                         const TUInt8Vec& runLengthBytes) {
+        return core::CPackedBitVector::readLastRunLength(lastRunBytes, runLengthBytes);
+    }
+    static std::size_t readRunLength(TUInt8VecCItr runLengthBytes) {
+        return core::CPackedBitVector::readRunLength(runLengthBytes);
+    }
+    static std::size_t popRunLength(TUInt8VecCItr& runLengthBytes) {
+        return core::CPackedBitVector::popRunLength(runLengthBytes);
+    }
+    static void writeRunLength(std::size_t runLength, TUInt8VecItr runLengthBytes) {
+        core::CPackedBitVector::writeRunLength(runLength, runLengthBytes);
+    }
+};
+}
+
+BOOST_AUTO_TEST_CASE(testInternals) {
+
+    using TUInt8Vec = std::vector<std::uint8_t>;
+
+    LOG_DEBUG(<< "bytes");
+
+    BOOST_TEST_REQUIRE(1, CPackedBitVectorInternals::bytes(1));
+    BOOST_TEST_REQUIRE(1, CPackedBitVectorInternals::bytes(
+                              CPackedBitVectorInternals::maximumOneByteRunLength()));
+    BOOST_TEST_REQUIRE(2, CPackedBitVectorInternals::bytes(
+                              CPackedBitVectorInternals::maximumOneByteRunLength() + 1));
+    BOOST_TEST_REQUIRE(2, CPackedBitVectorInternals::bytes(
+                              CPackedBitVectorInternals::maximumTwoByteRunLength()));
+    BOOST_TEST_REQUIRE(3, CPackedBitVectorInternals::bytes(
+                              CPackedBitVectorInternals::maximumTwoByteRunLength() + 1));
+    BOOST_TEST_REQUIRE(2, CPackedBitVectorInternals::bytes(
+                              CPackedBitVectorInternals::maximumThreeByteRunLength()));
+    BOOST_TEST_REQUIRE(3, CPackedBitVectorInternals::bytes(
+                              CPackedBitVectorInternals::maximumThreeByteRunLength() + 1));
```

Comment:
Is this correct? It looks like it should be compared to 3 and 4.

Reply:
No! The test assertion should, of course` have been `BOOST_REQUIRE_EQUAL` and the numbers are wrong! Good catch.

Reply:
Fixed in [018f0b3](https://github.com/elastic/ml-cpp/pull/1340/commits/018f0b3baa8390601ea1750f175d513778451883).